packages/atomic/src/components/result-lists/atomic-result-children/atomic-result-children.tsx

```diff
@@ -105,25 +124,47 @@ export class AtomicResultChildren {
     if (!content) return null;
     return (
       <atomic-result
+        key={child.result.uniqueId}
         content={content}
         result={child}
         engine={this.bindings.engine}
       ></atomic-result>
     );
   }
 
+  private getResult() {
+    return (
+      this.foldedResultListState.results.find((r) => {
```

Comment:
Similar as my above comment, this could be logic that we add to the headless controller for ease of use.